src/src/extra-providers.ts

```diff
@@ -8,23 +8,53 @@ import {
 
 @Injectable()
 export class SingleSpaPlatformLocation extends ɵBrowserPlatformLocation {
-  /**
-   * We could use the `inject` function from `@angular/core` which resolves
-   * dependencies from the currently active injector, but it's a feature
-   * of Angular 8+. Should be used when we drop support for older versions.
-   */
   private ngZone!: NgZone;
 
+  // This is a simple marker that helps us to ignore PopStateEvents
+  // that was not dispatched by the browser.
+  private skipNextPopState = false;
+
+  private onPopStateListeners: ((event: LocationChangeEvent) => void)[] = [];
+
+  destroy(): void {
+    // TLDR: Angular adds `popstate` event listener and then doesn't remove it when application gets destroyed.
+    // Basically, Angular has a potentional memory leak. The `ɵBrowserPlatformLocation`
+    // has `onPopState` method which adds `popstate` event listener and forgets, see here:
+    // https://github.com/angular/angular/blob/14be55c9facf3e47b8c97df4502dc3f0f897da03/packages/common/src/location/platform_location.ts#L126
+    for (const onPopStateListener of this.onPopStateListeners) {
+      window.removeEventListener('popstate', onPopStateListener);
+    }
+  }
+
+  pushState(state: any, title: string, url: string): void {
+    this.skipNextPopState = true;
+    super.pushState(state, title, url);
+  }
+
+  replaceState(state: any, title: string, url: string): void {
+    this.skipNextPopState = true;
+    super.replaceState(state, title, url);
+  }
+
   onPopState(fn: (event: LocationChangeEvent) => void): void {
-    super.onPopState(event => {
-      // Wrap any event listener into zone that is specific to some application.
-      // The main issue is `back/forward` buttons of browsers, because they invoke
-      // `history.back|forward` which dispatch `popstate` event. Since `single-spa`
-      // overrides `history.replaceState` Angular's zone cannot intercept this event.
-      // Only the root zone is able to intercept all events.
-      // See https://github.com/single-spa/single-spa-angular/issues/94 for more detail
-      this.ngZone.run(() => fn(event));
-    });
+    const onPopStateListener = (event: LocationChangeEvent) => {
+      if (this.skipNextPopState) {
```

Comment:
Does this approach cause issues when the back button is clicked? If so, try this?

```suggestion
      if (this.skipNextPopState && evt.singleSpa) {
```

More info at https://single-spa.js.org/docs/api#popstateevent